internal/eval/eval.go

```diff
@@ -646,6 +642,77 @@ func evalJSONGET(args []string, store *dstore.Store) []byte {
 	return clientio.Encode(string(resultBytes), false)
 }
 
+// evalJSONTOGGLE toggles a boolean value stored at the specified key and path.
+// args must contain at least the key and path (where the boolean is located).
+// If the key does not exist or is expired, it returns response.RespNIL.
+// If the field at the specified path is not a boolean, it returns an encoded error response.
+// If the boolean is `true`, it toggles to `false` (returns :0), and if `false`, it toggles to `true` (returns :1).
+// Returns an encoded error response if the incorrect number of arguments is provided.
+func evalJSONTOGGLE(args []string, store *dstore.Store) []byte {
+    if len(args) < 2 {
+        return diceerrors.NewErrArity("JSON.TOGGLE")
+    }
+
+    key := args[0]
+    path := args[1]
+
+    // Retrieve the object from the database
+    obj := store.Get(key)
+    if obj == nil {
+        return clientio.RespNIL
+    }
+
+    // Check if the object is of JSON type
+    errWithMessage := object.AssertTypeAndEncoding(obj.TypeEncoding, object.ObjTypeJSON, object.ObjEncodingJSON)
+    if errWithMessage != nil {
+        return errWithMessage
+    }
+
+    jsonData := obj.Value
+
+    // Parse the JSONPath expression
+    expr, err := jp.ParseString(path)
+    if err != nil {
+        return diceerrors.NewErrWithMessage("invalid JSONPath")
+    }
+
+    // Execute the JSONPath query
+    results := expr.Get(jsonData)
+    if len(results) == 0 {
+        return clientio.Encode([]interface{}{}, false)  // Return empty array if no matches
+    }
+
+    for _, result := range results {
+        switch v := result.(type) {
+        case bool:
+            newValue := !v
+            err := expr.Set(jsonData, newValue)
+            if err != nil {
+                return diceerrors.NewErrWithMessage("failed to set toggled value")
+            }
+
+            // Update the value in the store
+            newObj := &object.Obj{
+                Value:          jsonData,
+                TypeEncoding:   object.ObjTypeJSON,
+            }
+            store.Put(key, newObj)
+
+            // Return simple integer toggle result (1 for true, 0 for false)
+            if newValue {
+                return []byte(":1\r\n")
+            } else {
+                return []byte(":0\r\n")
+            }
+        default:
+            return []byte("-ERR Value at path is not a boolean\r\n")
+        }
```

Comment:
if the value is not bool, should return array that contains nil, also, because results may including many elements, you should return an array. just as the following:
`

      127.0.0.1:6379> JSON.SET user $ '{"age":13,"high":1.60,"flag":false,"pet":null,"language":["python","golang"],"flag":false,"partner":{"name":"tom","flag":true,"language":["rust"]}}'
    OK
    127.0.0.1:6379> JSON.TOGGLE user $.age
    1) (nil)
    127.0.0.1:6379> JSON.TOGGLE user $.flag
    1) (integer) 1
    127.0.0.1:6379> JSON.TOGGLE user $..flag
    1) (integer) 0
    2) (integer) 0

`